components/omega/test/infra/IOStreamTest.cpp

```diff
@@ -0,0 +1,285 @@
+   {
+
+      Calendar CalGreg("Gregorian", OMEGA::CalendarGregorian);
+      std::shared_ptr<Clock> ModelClock;
+
+      // Call initialization to create reference IO field
+      Err1 = initIOStreamTest(ModelClock, CalGreg);
+      TestEval("Initialize IOStream test", Err1, ErrRef, Err);
+
+      // Retrieve dimension lengths
+      I4 NCellsSize  = Dimension::getDimLengthLocal("NCells");
+      I4 NVertLevels = Dimension::getDimLengthLocal("NVertLevels");
+
+      // Create data arrays
+
+      Array2DR8 Temp("Temp", NCellsSize, NVertLevels);
+      Array2DR8 Salt("Salt", NCellsSize, NVertLevels);
+
+      // Attach data arrays to fields
+
+      Err1 = Field::attachFieldData<Array2DR8>("Temperature", Temp);
+      TestEval("Attach temperature data to field", Err1, ErrRef, Err);
+      Err1 = Field::attachFieldData<Array2DR8>("Salinity", Salt);
+      TestEval("Attach salinity data to field", Err1, ErrRef, Err);
+
+      // Validate all streams (Mesh stream already validated in HorzMesh?)
+      bool AllValidated = IOStream::validateAll();
+      TestEval("IOStream Validation", AllValidated, true, Err);
+
+      // Read restart file for initial temperature and salinity data
+      Metadata ReqMetadata; // leave empty for now - no required metadata
+      Err1 = IOStream::read("InitialState", *ModelClock, ReqMetadata);
+      TestEval("Read restart file", Err1, ErrRef, Err);
+
+      // Create a stop alarm at 1 year for time stepping
+      TimeInstant StopTime(&CalGreg, 0002, 1, 1, 0, 0, 0.0);
+      Alarm StopAlarm("Stop Time", StopTime);
+      Err1 = ModelClock->attachAlarm(&StopAlarm);
+      TestEval("Attach stop alarm", Err1, ErrRef, Err);
+
+      // Step forward in time and write files if it is time
+      while (!StopAlarm.isRinging()) {
+         ModelClock->advance();
+         TimeInstant CurTime    = ModelClock->getCurrentTime();
+         std::string CurTimeStr = CurTime.getString(4, 2, " ");
+
+         Err1 = IOStream::writeAll(*ModelClock);
+         if (Err1 != 0) // to prevent too much output in log
+            TestEval("Write all streams " + CurTimeStr, Err1, ErrRef, Err);
+      }
+
+      // Read one of the written files and check results
+      // Device arrays on edges
+      // parallelReduce(
+      //   {NEdgesSize}, KOKKOS_LAMBDA(int Edge, I4 &LCount) {
+      //      if (Data1DI8(Edge) != RefI8 + Edge) ++LCount;
+      //      if (*TPtr1DI8[Edge] != RefI8 + Edge) ++LCount;
+      //      for (int K = 0; K < NVertLevels; ++K) {
+      //         int Add = Edge*NVertLevels + K;
+      //         if (Data2DI8(Edge,K) != RefI8 + Edge + K) ++LCount;
+      //         if (Data2DR8(Edge,K) != RefR8 + Edge + K) ++LCount;
+      //         if (*Tptr2DI8[Add] != RefI8 + Edge + K) ++LCount;
+      //         if (*Tptr2DR8[Add] != RefR8 + Edge + K) ++LCount;
+      //      }
+      //   }
+      //   DataReducer
+      //);
+      // TestEval("Get data all edge device arrays, ptrs", DataCount1, 0, Err);
```

Comment:
This section of code commented out as well

Reply:
Yeah, I need to replace this with a read test.